### PR TITLE
Bug: fix the aa-kbc-params error in fedora

### PR DESCRIPTION
**Pull request by huoqifeng**
Fixes: #1852

### CODE REVIEW

**Comment by huoqifeng** [comment]
@mkulke @bpradipt not sure whether it's reasonable for azure and aws provider, may you please check?

**Comment by huoqifeng** [comment]
The problem is on fedora, the failure in `ExecStartPre` in `process-user-data` blocks  `ExecStart`.
I broken it into 2 services. `process-user-data-provision` and `process-user-data-update` while `process-user-data-update` depends on cloud-final.service because libvirt and other providers like ibmcloud uses cloud-init to provision user-data.

File: src/cloud-api-adaptor/podvm/files/etc/systemd/system/process-user-data-provision.service [hunk 1]
```diff
@@ -2,15 +2,14 @@
...
 [Unit]
-Description=Process user data
+Description=Process user data and provision
 After=network.target
 DefaultDependencies=no
 
 
 [Service]
 Type=oneshot
 ExecStartPre=/usr/local/bin/process-user-data provision-files
```
Comment (liudalibj) [inline src/cloud-api-adaptor/podvm/files/etc/systemd/system/process-user-data-provision.service:12]: Maybe `ExecStartPre` -> `ExecStart` here? It's a single service now.

**Comment by bpradipt** [comment]
@huoqifeng how was the libvirt provider working all this while with the packer created image?

File: src/cloud-api-adaptor/cmd/process-user-data/main.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ const (
 	providerAzure = "azure"
 	providerAws   = "aws"
 
-	defaultAgentConfigPath = "/etc/agent-config.toml"
+	defaultAgentConfigPath = "/run/peerpod/agent-config.toml"
```
Comment (liudalibj) [inline src/cloud-api-adaptor/cmd/process-user-data/main.go:22]: the defaultAgentConfigPath is changed here but
`src/cloud-api-adaptor/podvm/files/etc/systemd/system/kata-agent.service` didn't use the new changed path.

**Comment by huoqifeng** [comment]
> @huoqifeng how was the libvirt provider working all this while with the packer created image?

I'm not 100% sure, but I guess `ExecStartPre` failure won't cause `ExecStart` skipping on ubuntu.
Another thing I'm digging is why after the service change I made in this PR. the file `/etc/agent-config.toml` changed but not `/run/peerpod/agent-config.toml` without the main.go change. -- It worked on ubuntu image. @bpradipt

File: src/cloud-api-adaptor/cmd/process-user-data/main.go (hunk 2 above)
Comment (huoqifeng) [reply src/cloud-api-adaptor/cmd/process-user-data/main.go:22]: @liudalibj thanks for finding it, I think that's another reason why it works on ubuntu but not on fedora, fedora altered the path to `/run/peerpod/agent-config.toml` here https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf

Comment (huoqifeng) [reply src/cloud-api-adaptor/cmd/process-user-data/main.go:22]: I guess we should change to use `/run/peerpod/agent-config.toml` on ubuntu also.

Comment (huoqifeng) [reply src/cloud-api-adaptor/cmd/process-user-data/main.go:22]: I'll give a try after update the path...

Comment (liudalibj) [reply src/cloud-api-adaptor/cmd/process-user-data/main.go:22]: Agree, so for ubuntu this file need be updated in this pr, 

https://github.com/confidential-containers/cloud-api-adaptor/pull/1853/files#diff-7a67c51609d012acb98c931da3f33e5494e6d5722e99cd4d9e294ebcaf15b8faL10

File: src/cloud-api-adaptor/podvm-mkosi/mkosi.presets/initrd/mkosi.conf.d/fedora.conf [hunk 3]
```diff
@@ -3,7 +3,7 @@ Distribution=fedora
 
 [Distribution]
 Distribution=fedora
-Release=38
+Release=39
```
Comment (mkulke) [inline src/cloud-api-adaptor/podvm-mkosi/mkosi.presets/initrd/mkosi.conf.d/fedora.conf:6]: if we want to bump fedora, can we do this in a separate PR?

File: src/cloud-api-adaptor/podvm/files/etc/systemd/system/kata-agent.service [hunk 4]
```diff
@@ -1,15 +1,15 @@
 [Unit]
 Description=Kata Agent
 BindsTo=netns@podns.service
-Wants=process-user-data.service
-After=netns@podns.service process-user-data.service
+Wants=process-user-data-update.service
+After=netns@podns.service process-user-data-update.service
```
Comment (mkulke) [inline src/cloud-api-adaptor/podvm/files/etc/systemd/system/kata-agent.service:5]: regarding the idea discussed in https://github.com/confidential-containers/cloud-api-adaptor/issues/1852#issuecomment-2152135964, could we instead change the logic to provision the `agent-config.file` via cloud-config entry, so we don't have to perform any "updates"?

**Comment by huoqifeng** [comment]
I guess this has been fixed by PR https://github.com/confidential-containers/cloud-api-adaptor/pull/1858

**Comment by mkulke** [comment]
> I guess this has been fixed by PR #1858 @mkulke

I wouldn't expect this (at least intentionally). the above PR should not change behaviour in this regard.

**Comment by huoqifeng** [comment]
Close it as it has been fixed in https://github.com/confidential-containers/cloud-api-adaptor/pull/1868
